doc/design/ib-vf-configuration.md

```diff
@@ -0,0 +1,72 @@
+---
+title: IB VF GUID Configuration
+authors:
+  - almaslennikov
+reviewers:
+  - SchSeba
+  - adrianchiris
+creation-date: 11-03-2024
+last-updated: 11-03-2024
+---
+
+# IB VF GUID Configuration
+
+## Summary
+Allow SR-IOV Network Operator to use a static configuration file from the host filesystem to assign GUIDs to IB VFs
+
+## Motivation
+We have customers using the SR-IOV operator to create IB VFs, and they need a way to automate GUID assignment,
+so that IB VFs are automatically bound to the required PKeys and no additional manual configuration is needed.
+In this use case, the GUID configuration is static and known in advance.
+Now the GUIDs are assigned by the sriov-network-config-daemon randomly.
```

Comment:
Can you mention in this section the reason why we need to handle this in the operator and not in the CNI?

Reply:
Added new sections on an alternative solution and the comparison between the two